lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2547,6 +2547,13 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       }
     }
 
+    // -----
+    // NonCopyableChecks
+    //
+
+    if (ED->isObjC() && ED->isMoveOnly()) {
+      ED->diagnose(diag::moveonly_objc_enum_banned);
```

Comment:
Can you merge this with the checks on line 2560 below?

Reply:
I am happy to do that in a subsequent commit. Can you be more specific... about what you want?